Database documentation/MongoDB.md

```diff
@@ -0,0 +1,46 @@
+# Database
+
+## MongoDB
+
+MongoDB is een open source document-georiënteerde database en is geschreven in C++. Er is geen schema, de documenten worden in de vorm van BSON (binair JSON) opgeslagen en de structuur van deze documenten is flexibel. De database kan gemakkelijk gedistribueerd worden, de data wordt dan over meerdere computers verspreid om gedistribueerde gegevensverwerking mogelijk te maken. MongoDB is geen relationeel databasemanagementsysteem, er is geen ondersteuning voor joins en voldoet ook niet aan de ACID-regels want de ondersteuning voor transacties is beperkt. MongoDB wordt gerekend tot de zogenaamde NoSQL-databases.
+
+MongoDb is geschikt voor grote data het is schaalbaar en werk zonder relaties in tegenstelling tot bijvoorbeeld SQLite. Omdat we in ons project geen echte relaties nodig hebben is de keuze daarom ook makkelijk gemaakt voor MongoDB.
+
+### Voorbeeld eventuele database structuur
+
+```json
+{
+ "user_id": "1",
+ "username": "Maliek Meersschaert",
+ "password": "SOME HASH",
+ "polls": [
+     {
+      "poll_id": "1",
+      "name": "Amerikaanse verkiezingen",
+      "options": ["Donald Trump", "Hilary Clinton", "Gary Johnson", "Jill Stein"],
+      "values": [10,9,2,1]
+    },
+    {
+      "poll_id": "2",
+      "name": "Lievelingskleur",
+      "options": ["Geel", "Blauw"],
+      "values": [1,2]
+    }
+  ]
+}
+ ```
```

Comment:
Hier zou ik een andere structuur gebruiken: 

```json
{
  "polls": {
    "1": {
      "name": "Amerikaanse verkiezingen",
      "options": {
        "Donald Trump": 10,
        "Hilary Clinton": 9,
        "Gary Johnson": 2,
        "Jill Stein": 1
      }
    },
    "2": {
      "name": "Lievelingskleur",
      "options": {
        "Geel": 1,
        "Blauw": 2
      },
    },
    "3": {
      "name": "Kom je naar mijn feestje?",
      "options": {
        "ja": 30,
        "nee": 5
      }
    }
  },
  "users": {
    "useridofmaliek": {
      "username": "Maliek Meersschaert",
      "password": "SOME HASH",
      "polls": {
        "1": true,
        "2": true
      }
    }
  }
}
```

Waarom? omdat niet-ingelogde gebruikers ook een poll kunnen hebben, dus moeten polls apart.

Een ander idee is om wel de poll-objecten volledig onder de gebruiker te doen, en een anonieme gebruiker aan te maken voor elke poll die niet ingelogd gemaakt is. Dan zou je er via paul.gent/gebruikeranoniem/pollid naar toe kunnen gaan, waarbij die random uuid's toegekend worden (met iets zoals https://github.com/Haroenv/NewBase60)